compiler/rustc_trait_selection/src/traits/chalk_fulfill.rs

```diff
@@ -58,6 +58,9 @@ impl TraitEngine<'tcx> for FulfillmentContext<'tcx> {
                     obligation: obligation.clone(),
                     code: FulfillmentErrorCode::CodeAmbiguity,
                     points_at_arg_span: false,
+                    // FIXME - does Chalk have a notation of 'root obligation'?
```

Comment:
cc @rust-lang/wg-traits: Does Chalk have a way of getting the obligation first passed to `TraitEngine::register_predicate_obligation`?

Reply:
Well, does *Chalk*? No. Chalk only takes a goal and produces an answer. Does the *chalk_fulfill* code? I'm not sure, I'll have to go back through this.

I imagine that the code here is most correct. The rustc trait solver will register predicates as it tries to solve a predicate, whereas Chalk *won't*, it just gives back an answer.

Reply:
Yeah, chalk sort of *only* has a notion of root obligation, at least as presently designed.

Reply:
I'm not too concerned because I already fully expect to have to rework a lot of things when we try to land chalk :)